ProcessLib/ThermoMechanics/ThermoMechanicsFEM.h

```diff
@@ -285,10 +295,11 @@ class ThermoMechanicsLocalAssembler
             // assume isotropic thermal expansion
             const double T_ip = N.dot(T);  // T at integration point
             eps_m.noalias() =
-                eps - linear_thermal_strain_increment * Invariants::identity2;
+                eps_m_prev + eps - eps_prev -
+                linear_thermal_strain_increment * Invariants::identity2;
+
             auto&& solution = _ip_data[ip].solid_material.integrateStress(
```

Comment:
For the stress integration, I think we only need two variables of strain, i.e. the increment of strain that including thermal strain increment, and the mechanical strain of the previous time step for some special material model. I will try it in future. In the current structure of the stress integration, the changes presented in the PR is quite good.

Reply:
But in order that `pushBackState()` can update the state maybe one has to keep four strain variables anyway.